src/twisted/conch/ssh/transport.py

```diff
@@ -1036,6 +1087,20 @@ def sendDisconnect(self, reason, desc):
         )
         self.transport.loseConnection()
 
+    def sendExtInfo(self, extensions):
+        """
+        Send an RFC 8308 extension advertisement to the client, if permitted.
+
```

Comment:
I think that this can also be used by the client to advertise its extensions.

Or if in twisted this should only be used by the server side, maybe move it to server code and make it private.

```suggestion
    def sendExtInfo(self, extensions):
        """
        Send an RFC 8308 extension advertisement to the remote peer.
        
        Nothing is sent if the peer doesn't support negotiations.
```

and if this should be used by the client side, I think that we should expose `_supportsExtensionNegotiation` as a private member so that the an external code using the client can query if the server supports any extensions.

Reply:
Your suggestion looks good; I've applied it, thanks.

I don't have a concrete use case for using this from the client side at the moment, so I'd prefer to keep the attribute private for now to avoid accidentally painting ourselves into a corner by making it public.  No problem with making it public later if that turns out to be useful.